build_tools/on_pr_comment_update_environments_and_lock_files.py

```diff
@@ -0,0 +1,67 @@
+import argparse
+import os
+import shlex
+import subprocess
+
+
+def execute_command(command):
+    command_list = shlex.split(command)
+    subprocess.run(command_list, check=True, text=True)
+
+
+def main():
+    comment = os.environ["COMMENT"].splitlines()[0].strip()
+
+    # Extract the command-line arguments from the comment
+    prefix = "@scikit-learn-bot update lock-files"
+    assert comment.startswith(prefix)
+    all_args_list = shlex.split(comment[len(prefix) :])
+
+    # Parse the options for the lock-file script
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--select-build", default="")
+    parser.add_argument("--skip-build", default=None)
+    parser.add_argument("--select-tag", default=None)
+    args, extra_args_list = parser.parse_known_args(all_args_list)
+
+    # Rebuild the command-line arguments for the lock-file script
+    args_string = ""
+    if args.select_build != "":
+        args_string += f" --select-build {args.select_build}"
+    if args.skip_build is not None:
+        args_string += f" --skip-build {args.skip_build}"
+    if args.select_tag is not None:
+        args_string += f" --select-tag {args.select_tag}"
+
+    # Parse extra arguments
+    extra_parser = argparse.ArgumentParser()
+    extra_parser.add_argument("--commit-marker", default=None)
+    extra_args, _ = extra_parser.parse_known_args(extra_args_list)
+
+    marker = ""
+    # Additional markers based on the tag
+    if args.select_tag == "main-ci":
+        marker += "[doc build] "
+    elif args.select_tag == "scipy-dev":
+        marker += "[scipy-dev] "
+    elif args.select_tag == "arm":
+        marker += "[cirrus arm] "
+    elif len(all_args_list) == 0:
+        # No arguments which will update all lock files so add all markers
+        marker += "[doc build] [scipy-dev] [cirrus arm] "
+    # The additional `--commit-marker` argument
+    if extra_args.commit_marker is not None:
+        marker += extra_args.commit_marker + " "
+
+    execute_command(
+        f"python build_tools/update_environments_and_lock_files.py{args_string}"
+    )
+    execute_command('git config --global user.name "scikit-learn-bot"')
+    execute_command('git config --global user.email "noreply@github.com"')
+    execute_command("git add -A")
+    execute_command(f'git commit -m "{marker}Update lock files"')
+    execute_command("git push")
```

Comment:
maybe here we should make sure we're not pushing to the main repo? As in, check `remote` and make sure it's not `scikit-learn` and it's the same as the users?

Reply:
IMHO it's not necessary because I think we can trust the official `actions/checkout` that when pushing it would always respect the repository and ref we checked out, which as mentioned in https://github.com/scikit-learn/scikit-learn/pull/29505#discussion_r1713170424 will (theoretically) be the PR branch in the contributor's fork. Perhaps we can instead check that `pr_head_repository` and `pr_head_ref` are not `scikit-learn/scikit-learn` and `main` in an earlier step?